i915.c

```diff
@@ -560,16 +555,40 @@ static void *i915_bo_map(struct bo *bo, struct vma *vma, size_t plane, uint32_t
 
 		addr = (void *)(uintptr_t)gem_map.addr_ptr;
 	} else {
-
 		struct drm_i915_gem_mmap_gtt gem_map;
 		memset(&gem_map, 0, sizeof(gem_map));
 
 		gem_map.handle = bo->handles[0].u32;
 
 		ret = drmIoctl(bo->drv->fd, DRM_IOCTL_I915_GEM_MMAP_GTT, &gem_map);
 		if (ret) {
-			drv_log("DRM_IOCTL_I915_GEM_MMAP_GTT failed\n");
-			return MAP_FAILED;
+			struct driver *drv = bo->drv;
+			if ((drv->gpu_grp_type == TWO_GPU_IGPU_VIRTIO) || (drv->gpu_grp_type == THREE_GPU_IGPU_VIRTIO_DGPU)) {
+				struct drm_i915_gem_mmap gem_map;
+				memset(&gem_map, 0, sizeof(gem_map));
+				if ((bo->meta.use_flags & BO_USE_SCANOUT) &&
+				    !(bo->meta.use_flags &
+				      (BO_USE_RENDERSCRIPT | BO_USE_CAMERA_READ | BO_USE_CAMERA_WRITE)))
+					gem_map.flags = I915_MMAP_WC;
+				gem_map.handle = bo->handles[0].u32;
+				gem_map.offset = 0;
+				gem_map.size = bo->meta.total_size;
+				ret = drmIoctl(bo->drv->fd, DRM_IOCTL_I915_GEM_MMAP, &gem_map);
+				if (ret) {
+				        drv_log("DRM_IOCTL_I915_GEM_MMAP failed\n");
+				        return MAP_FAILED;
+				}
+				addr = (void *)(uintptr_t)gem_map.addr_ptr;
+				if (addr == MAP_FAILED) {
```

Comment:
such check seems incorrect

Reply:
it is same check as http://inlubt0232.iind.intel.com:8080/source/xref/CELADON/hardware/intel/external/minigbm-intel/i915.c#539